src/test/regress/sql/multi_create_table_constraints.sql

```diff
@@ -0,0 +1,208 @@
+);
+SELECT master_create_distributed_table('pk_on_part_col_named', 'partition_col', 'hash');
+
+CREATE TABLE uq_part_col_named
+(
+	partition_col integer CONSTRAINT uq_part_col_named_uniq UNIQUE,
+	other_col integer
+);
+SELECT master_create_distributed_table('uq_part_col_named', 'partition_col', 'hash');
+
+CREATE TABLE uq_two_columns_named
+(
+	partition_col integer,
+	other_col integer,
+	CONSTRAINT uq_two_columns_named_uniq UNIQUE (partition_col, other_col)
+);
+SELECT master_create_distributed_table('uq_two_columns_named', 'partition_col', 'hash');
+SELECT master_create_worker_shards('uq_two_columns_named', '4', '2');
+INSERT INTO uq_two_columns_named (partition_col, other_col) VALUES (1,1);
+INSERT INTO uq_two_columns_named (partition_col, other_col) VALUES (1,1);
+
+CREATE TABLE ex_on_part_col_named
+(
+	partition_col integer,
+	other_col integer,
+	CONSTRAINT ex_on_part_col_named_exclude EXCLUDE (partition_col WITH =)
+);
+SELECT master_create_distributed_table('ex_on_part_col_named', 'partition_col', 'hash');
+SELECT master_create_worker_shards('ex_on_part_col_named', '4', '2');
+INSERT INTO ex_on_part_col_named (partition_col, other_col) VALUES (1,1);
+INSERT INTO ex_on_part_col_named (partition_col, other_col) VALUES (1,2);
+
+CREATE TABLE ex_on_two_columns_named
+(
+	partition_col integer,
+	other_col integer,
+	CONSTRAINT ex_on_two_columns_named_exclude EXCLUDE (partition_col WITH =, other_col WITH =)
+);
+SELECT master_create_distributed_table('ex_on_two_columns_named', 'partition_col', 'hash');
+SELECT master_create_worker_shards('ex_on_two_columns_named', '4', '2');
+INSERT INTO ex_on_two_columns_named (partition_col, other_col) VALUES (1,1);
+INSERT INTO ex_on_two_columns_named (partition_col, other_col) VALUES (1,1);
+
+CREATE TABLE ex_wrong_operator_named
+(
+	partition_col tsrange,
+	other_col tsrange,
+	CONSTRAINT ex_wrong_operator_named_exclude EXCLUDE USING gist (other_col WITH =, partition_col WITH &&)
+);
+SELECT master_create_distributed_table('ex_wrong_operator_named', 'partition_col', 'hash');
+
+CREATE TABLE ex_overlaps_named
+(
+	partition_col tsrange,
+	other_col tsrange,
+	CONSTRAINT ex_overlaps_operator_named_exclude EXCLUDE USING gist (other_col WITH &&, partition_col WITH =)
+);
+SELECT master_create_distributed_table('ex_overlaps_named', 'partition_col', 'hash');
+SELECT master_create_worker_shards('ex_overlaps_named', '4', '2');
+INSERT INTO ex_overlaps_named (partition_col, other_col) VALUES ('[2016-01-01 00:00:00, 2016-02-01 00:00:00]', '[2016-01-01 00:00:00, 2016-02-01 00:00:00]');
+INSERT INTO ex_overlaps_named (partition_col, other_col) VALUES ('[2016-01-01 00:00:00, 2016-02-01 00:00:00]', '[2016-01-15 00:00:00, 2016-02-01 00:00:00]');
+
+-- now show that Citus allows unique constraints on range-partitioned tables.
+
+CREATE TABLE uq_range_tables
+(
+	partition_col integer UNIQUE,
+	other_col integer
+);
+SELECT master_create_distributed_table('uq_range_tables', 'partition_col', 'range');
```

Comment:
It can be beneficial to include some more tests about constraints with IMMUTABLE functions and partial constraints.

Adding a CHECK constraint test would also be nice but it is not directly related to this PR.